bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -355,6 +358,7 @@ public Image(Device device, Rectangle bounds) {
 	bounds = DPIUtil.scaleUp (bounds, getZoom());
 	init(bounds.width, bounds.height);
 	init();
+	this.device.registerResourceWithZoomSupport(this);
```

Comment:
init seem to be called everywhere where this code is called, so maybe put it in the init method?!?

Reply:
We only wanna register those resources which have multiple handles for different zoom levels. But some other classes like Cursor, Font, etc also call init() but they don't have multiple handles like Image, Path, Pattern, etc so it would be better to call it from the constructors. Also, some constructors like Image(Device device) do not call init() so it will make the registration of such resources inconsistent.

Reply:
Are you satisfied with the answer @laeubi ? May we mark this conversation as "resolved"?

Reply:
@laeubi since we would like to have this in M2 and the current approach LGTM, I merged this PR. 

We can continue this discussion in a follow-up PR if necessary.